### ISSUE TITLE
[Feature/Addon Request] Set consumed shaders via custom item component

### ISSUE THREAD

**Issue by MCLegoMan** [Nettakrim/Souper-Secret-Settings#14]
Note: this feature would require the mod to be installed on both the client and the server.
*could it work better as an addon mod? or part of soup but have it optional for servers?*

1.20.5 added Item Components, using these we could make any consumable item give a shader(s) when consumed.
Ideally, we could make it so when the player used the following command: `/give @a minecraft:bread[souper_secret_settings:shaders={shaders:['minecraft:creeper','minecraft:invert']}]`, and then consumed the bread they'd get both the creeper and invert shaders.

If the item component is on soup, it should use the item component instead of a random shader.
If the item component is provided but empty, it could use a random shader.

**Comment by Nettakrim** [comment]
ooh yeah that is interesting, i need to look into components some more since they do seem cool

**Comment by MCLegoMan** [comment]
yeah, I used them in my [april fools mod](https://github.com/MCLegoMan/mclmaf2024) pretty much as they released, they're quite cool.

**Comment by Nettakrim** [comment]
this might be the sort of feature that would make more sense to be part of https://modrinth.com/mod/client-execution

**Comment by mclegoman** [comment]
> this might be the sort of feature that would make more sense to be part of https://modrinth.com/mod/client-execution 

Would make sense to be honest.

**Comment by Nettakrim** [comment]
if i ever do something like this, it will be as part of a separate mod - either client execution, or something else ive been thinking of
